components/tyk-dashboard/templates/deployment-dashboard.yaml

```
@@ -0,0 +1,253 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dashboard-{{ include "tyk-dashboard.fullname" . }}
+  labels:
+    app: dashboard-{{ include "tyk-dashboard.fullname" . }}
+    chart: {{ include "tyk-dashboard.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.dashboard.replicaCount }}
+  minReadySeconds: 5
+  strategy:
+    # indicate which strategy we want for rolling update
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 2
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: dashboard-{{ include "tyk-dashboard.fullname" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: dashboard-{{ include "tyk-dashboard.fullname" . }}
+        release: {{ .Release.Name }}
+      annotations:
+      {{- if .Values.dashboard.enableIstioIngress }}
+        traffic.sidecar.istio.io/excludeInboundPorts: "{{ .Values.dashboard.containerPort }}"
+        traffic.sidecar.istio.io/includeInboundPorts: ""
+      {{- end }}
+        checksum/secrets: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- if .Values.mounts }}
+        {{- range $secret := .Values.mounts }}
+        "checksum/secret-{{ $secret.name }}": {{ include (print "mounts/%s" $secret.filename) . | sha256sum }}
+        {{- end }}
+        {{- end }}
+    spec:
+{{- if .Values.dashboard.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.dashboard.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.dashboard.tolerations }}
+      tolerations:
+{{ toYaml .Values.dashboard.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.dashboard.affinity }}
+      affinity:
+{{ toYaml .Values.dashboard.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - image: "{{ .Values.dashboard.image.repository }}:{{ .Values.dashboard.image.tag }}"
+        imagePullPolicy: {{ .Values.dashboard.image.pullPolicy }}
+        name: dashboard-{{ .Chart.Name }}
+        securityContext:
+          runAsUser: 999
+          runAsGroup: 999
+        env:
+          ## DEFAULT TYK DASHBOARD CONFIGURATION
+          - name: "TYK_DB_ENABLEOWNERSHIP"
+            value: "{{ .Values.dashboard.enableOwnership }}"
+          - name: "TYK_DB_LOGLEVEL"
+            value: "debug"
+          - name: "TYK_LOGLEVEL"
+            value: "debug"
+          - name: "TYK_DB_PAGESIZE"
+            value: "{{ .Values.dashboard.defaultPageSize }}"
+          - name: TYK_GW_STORAGE_ADDRS
```

```suggestion
          - name: TYK_DB_REDISADDRS
```
We are setting this environment variable below. So we don't need this actually